editor/js/Viewport.js

```diff
@@ -285,19 +284,19 @@ var Viewport = function ( editor ) {
 
 	signals.transformModeChanged.add( function ( mode ) {
 
-		transformControls.setMode( mode );
```

Comment:
I think I prefer keeping these methods, it's easier to keep backwards compatibility when changing the API...

Reply:
I was hoping to maintain TransformControls as something integral to an editor built with my custom UI framework.

This framework relies heavily on a data binding implementation which avoids setter functions by design.

I have nothing against having specialized setter functions as long as the object.defined property getters/setters work as designed here. I can just remove deprication warnings and leave the functions hanging there... Sounds good?

Reply:
@mrdoob I don't know if you've noticed by the way. The functions are still there but throw a deprication warning.